dev/DynamicDependency/M.AM.DD.PackageDependency.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "M.AM.DD.PackageDependency.h"
+#include "Microsoft.ApplicationModel.DynamicDependency.PackageDependency.g.cpp"
+
+#include "M.AM.DD.PackageDependencyContext.h"
+
+#include <MsixDynamicDependency.h>
+#include <wil/resource.h>
+#include <wil_msixdynamicdependency.h>
+
+#include <M.AM.Converters.h>
+
+namespace winrt::Microsoft::ApplicationModel::DynamicDependency::implementation
+{
+    PackageDependency::PackageDependency(PCWSTR id)
+    {
+        m_id = winrt::to_hstring(id);
+    }
+    PackageDependency::PackageDependency(hstring const& id) :
+        m_id(id)
+    {
+    }
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::GetFromId(hstring const& id)
+    {
+        return winrt::make<implementation::PackageDependency>(id);
+    }
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::Create(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& minVersion)
+    {
+        const auto mddMinVersion = ::Microsoft::ApplicationModel::DynamicDependency::ToVersion(minVersion);
+        const auto mddArchitectures = MddPackageDependencyProcessorArchitectures::None;
+        const auto mddLifetimeKind = MddPackageDependencyLifetimeKind::Process;
+        const auto mddOptions = MddCreatePackageDependencyOptions::None;
+        wil::unique_process_heap_string packageDependencyId;
+        THROW_IF_FAILED(MddTryCreatePackageDependency(nullptr, packageFamilyName.c_str(), mddMinVersion, mddArchitectures, mddLifetimeKind, nullptr, mddOptions, wil::out_param(packageDependencyId)));
+        return GetFromId(packageDependencyId.get());
+    }
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::Create(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& minVersion, Microsoft::ApplicationModel::DynamicDependency::CreatePackageDependencyOptions const& options)
+    {
+        const auto mddMinVersion = ::Microsoft::ApplicationModel::DynamicDependency::ToVersion(minVersion);
```

Comment:
These three should probably call some worker internal method, instead of repeating the same code over and over?